gap_linux.go

```diff
@@ -496,3 +544,74 @@ func (a *Adapter) SetRandomAddress(mac MAC) error {
 
 	return nil
 }
+
+func (a *Advertisement) handleDBusSignals() {
+	for {
+		select {
+		case sig, ok := <-a.sigCh:
+			if !ok {
+				return // channel closed
+			}
+
+			if sig.Name != "org.freedesktop.DBus.Properties.PropertiesChanged" {
+				continue
+			}
+
+			// Skip any signals that are not the Device1 interface.
+			if interfaceName, ok := sig.Body[DBusPropertiesChangedInterfaceName].(string); !ok || interfaceName != "org.bluez.Device1" {
+				continue
+			}
+
+			// Get all changed properties and skip any signals that are not
+			// compliant with the Device1 interface.
+			changes, ok := sig.Body[DBusPropertiesChangedDictionary].(map[string]dbus.Variant)
+			if !ok {
+				continue
+			}
+
+			// Call the connect handler if the Connected property has changed.
+			if connected, ok := changes["Connected"].Value().(bool); ok {
```

Comment:
I wonder if this is not completely correct? Seems like the first time that a device connects, since there is no value, there is no call to the connect handler. The following disconnect, and subsequent connect/disconnects, do work correctly.

See the following output:

```
$ go run ./examples/advertisement/
advertising...
device disconnected: 59:5E:7D:4D:4B:09

$ go run ./examples/advertisement/
advertising...
device connected: 59:5E:7D:4D:4B:09
device disconnected: 59:5E:7D:4D:4B:09
```

What do you think?

Reply:
Confirmed that it is now working as expected. Thanks!